tests/gpflow/conditionals/test_multioutput.py

```diff
@@ -317,29 +317,104 @@ def test_sample_conditional_mixedkernel():
     )
 
 
-@pytest.mark.parametrize(
-    "func, R",
-    [
-        (fully_correlated_conditional_repeat, 5),
-        (fully_correlated_conditional_repeat, 1),
-        (fully_correlated_conditional, 1),
-    ],
+@pytest.fixture(
+    name="q_sqrt_factory", params=[lambda _, __: None, lambda LM, R: tf.eye(LM, batch_shape=(R,))]
 )
-def test_fully_correlated_conditional_repeat_shapes(func, R):
+def _q_sqrt_factory_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="full_cov", params=[True, False])
+def _full_cov_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="full_output_cov", params=[True, False])
+def _full_output_cov_fixture(request):
+    return request.param
+
+
+@pytest.mark.parametrize("R", [1, 2, 5])
+def test_fully_correlated_conditional_repeat_shapes_fc_and_foc(
+    R, q_sqrt_factory, full_cov, full_output_cov
+):
+    # fully_correlated_conditional_repeat does not support whiten = False
+    whiten = True
```

Comment:
Assuming that the lack of whiten=False support is a lack of implementation, not an intrinsic impossibility:
```suggestion
@pytest.mark.parametrize("R", [1, 2, 5])
@pytest.mark.parametrize("whiten", [
    True,
    pytest.param(
        False,
        marks=pytest.mark.xfail("fully_correlated_conditional_repeat does not support whiten=False"),
    ),
])       
def test_fully_correlated_conditional_repeat_shapes_fc_and_foc(
    R, q_sqrt_factory, full_cov, full_output_cov, whiten
):
```
(will probably require a `make format`)